### PR TITLE
Check eligibility upon submit

### LINKED ISSUE #4080: Eligibility: Validate application can be submitted upon submit
The applicant should be able to attempt to submit and ineligible application and should be redirected to the NotEligibleView

### PR DESCRIPTION
**Pull request by shanemc-goog**
### Description

When an application is ineligible, applicants can still submit but they are instead shown the Ineligible page view which contains more information.

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

#### New Features

- [ ] Add new FeatureFlag env vars to `server/conf/application.conf`
- [ ] Conditioned new functionality on a [FeatureFlag](https://docs.civiform.us/contributor-guide/developer-guide/feature-flags)
- [ ] Wrote browser tests with the feature flag off and on, etc.

### Issue(s) this completes

Fixes #4080

### CODE REVIEW

File: server/app/controllers/applicant/ApplicantProgramReviewController.java [hunk 1]
```diff
@@ -165,6 +178,20 @@ private CompletionStage<Result> submitInternal(
                       routes.ApplicantProgramReviewController.review(applicantId, programId);
                   return redirect(reviewPage).flashing("error", errorMsg);
                 }
+                if (cause instanceof ApplicationNotEligibleException) {
```
Comment (bion) [inline server/app/controllers/applicant/ApplicantProgramReviewController.java:181]: thought: how did the official java way to do async programming not get feature parity with `catch` for handling typed exceptions? smh

File: server/test/services/applicant/ApplicantServiceTest.java [hunk 2]
```diff
@@ -31,17 +31,12 @@
 import services.LocalizedStrings;
 import services.Path;
 import services.applicant.ApplicantService.ApplicantProgramData;
-import services.applicant.exception.ApplicantNotFoundException;
-import services.applicant.exception.ApplicationOutOfDateException;
-import services.applicant.exception.ApplicationSubmissionException;
-import services.applicant.exception.ProgramBlockNotFoundException;
+import services.applicant.exception.*;
```
Comment (bion) [inline server/test/services/applicant/ApplicantServiceTest.java:34]: issue (blocking): no wildcard imports

File: server/test/services/applicant/ApplicantServiceTest.java [hunk 3]
```diff
@@ -31,17 +31,12 @@
...
 import services.applicant.question.Scalar;
 import services.application.ApplicationEventDetails;
 import services.application.ApplicationEventDetails.StatusEvent;
-import services.program.PathNotInBlockException;
-import services.program.ProgramDefinition;
-import services.program.ProgramNotFoundException;
-import services.program.StatusDefinitions;
+import services.program.*;
+import services.program.predicate.*;
```
Comment (bion) [inline server/test/services/applicant/ApplicantServiceTest.java:39]: issue (blocking): no wildcard imports

**Review by bion — APPROVED** [review]

File: server/test/services/applicant/ApplicantServiceTest.java (hunk 2 above)
Comment (shanemc-goog) [reply server/test/services/applicant/ApplicantServiceTest.java]: done.  My IntelliJ is auto switching to * imports on the 5th in a package it seems.... Will have to figure out how to disable that.

File: server/test/services/applicant/ApplicantServiceTest.java (hunk 3 above)
Comment (shanemc-goog) [reply server/test/services/applicant/ApplicantServiceTest.java]: done

File: server/app/controllers/applicant/ApplicantProgramReviewController.java (hunk 1 above)
Comment (shanemc-goog) [reply server/app/controllers/applicant/ApplicantProgramReviewController.java:181]: Clean use of Futures in java is def a weakness for me, not sure if there's a better way to manage this.  We also have to make them Runtime exceptions even if they maybe shouldn't it seems.

Comment (bion) [reply server/app/controllers/applicant/ApplicantProgramReviewController.java:181]: I don't know of a better way to handle this, I think it's an oversight in the language design

**Review by bion — APPROVED** [review]

**Pull request merged**